windows/CodePush.cs

```diff
@@ -0,0 +1,668 @@
+        private readonly string REACT_DEV_BUNDLE_CACHE_FILE_NAME = "ReactNativeDevBundle.js";
+
+        // Helper classes.
+        private CodePushNativeModule codePushNativeModule;
+        private CodePushPackage codePushPackage;
+
+        // Config properties.
+        private string appVersion;
+        private string deploymentKey;
+        private readonly string serverUrl = "https://codepush.azurewebsites.net/";
+
+        private ReactPage mainPage;
+
+        private static CodePushModule currentInstance;
+
+        public CodePushModule(string deploymentKey, ReactPage mainPage)
+        {
+            codePushPackage = new CodePushPackage();
+            // TODO implement telemetryManager 
+            // this.codePushTelemetryManager = new CodePushTelemetryManager(this.applicationContext, CODE_PUSH_PREFERENCES);
+            this.deploymentKey = deploymentKey;
+            this.mainPage = mainPage;
+            appVersion = Package.Current.Id.Version.Major + "." + Package.Current.Id.Version.Minor + "." + Package.Current.Id.Version.Build;
+            InitializeUpdateAfterRestart();
+            if (currentInstance != null)
+            {
+                CodePushUtils.log("More than one CodePush instance has been initialized. Please use the instance method codePush.getBundleUrlInternal() to get the correct bundleURL for a particular instance.");
+            }
+
+            currentInstance = this;
+        }
+
+        private async Task ClearReactDevBundleCache()
+        {
+            StorageFile devBundleCacheFile = null;
+            try
+            {
+                devBundleCacheFile = await ApplicationData.Current.LocalFolder.GetFileAsync(REACT_DEV_BUNDLE_CACHE_FILE_NAME);
+            }
+            catch (FileNotFoundException)
+            {
+            }
+
+            if (devBundleCacheFile != null)
+            {
+                await devBundleCacheFile.DeleteAsync();
+            }
+        }
+
+        private async Task<long> GetBinaryResourcesModifiedTime()
+        {
+            StorageFile assetJSBundleFile = await StorageFile.GetFileFromApplicationUriAsync(new Uri(ASSETS_BUNDLE_PREFIX + assetsBundleFileName));
+            BasicProperties fileProperties = await assetJSBundleFile.GetBasicPropertiesAsync();
+            return fileProperties.DateModified.ToUnixTimeMilliseconds();
+        }
+
+        public string GetJavaScriptBundleFile()
+        {
+            return GetJavaScriptBundleFile(DEFAULT_JS_BUNDLE_NAME);
+        }
+
+        public string GetJavaScriptBundleFile(string assetsBundleFileName)
+        {
+            if (currentInstance == null)
+            {
+                throw new CodePushNotInitializedException("A CodePush instance has not been created yet. Have you added it to your app's list of ReactPackages?");
+            }
```

Comment:
We should be able to just throw an [InvalidOoerationExceprion](https://msdn.microsoft.com/en-us/library/system.invalidoperationexception(v=vs.110).aspx)